Sources/CocoaLumberjack/DDFileLogger.m

```diff
@@ -1526,145 +1498,8 @@ - (void)renameFile:(NSString *)newFileName {
-    }
-
-    return NO;
-}
-
-- (void)addExtensionAttributeWithName:(NSString *)attrName {
-    // This method is only used on the iPhone simulator, where normal extended attributes are broken.
-    // See full explanation in the header file.
-
-    if ([attrName length] == 0) {
-        return;
-    }
-
-    // Example:
-    // attrName = "archived"
-    //
-    // "mylog.txt" -> "mylog.archived.txt"
-    // "mylog"     -> "mylog.archived"
-
-    NSArray *components = [[self fileName] componentsSeparatedByString:@"."];
-
-    NSUInteger count = [components count];
-
-    NSUInteger estimatedNewLength = [[self fileName] length] + [attrName length] + 1;
-    NSMutableString *newFileName = [NSMutableString stringWithCapacity:estimatedNewLength];
-
-    if (count > 0) {
-        [newFileName appendString:components.firstObject];
-    }
-
-    NSString *lastExt = @"";
-
-    NSUInteger i;
-
-    for (i = 1; i < count; i++) {
-        NSString *attr = components[i];
-
-        if ([attr length] == 0) {
-            continue;
-        }
-
-        if ([attrName isEqualToString:attr]) {
-            // Extension attribute already exists in file name
-            return;
-        }
-
-        if ([lastExt length] > 0) {
-            [newFileName appendFormat:@".%@", lastExt];
-        }
-
-        lastExt = attr;
-    }
-
-    [newFileName appendFormat:@".%@", attrName];
-
-    if ([lastExt length] > 0) {
-        [newFileName appendFormat:@".%@", lastExt];
-    }
-
-    [self renameFile:newFileName];
-}
-
-- (void)removeExtensionAttributeWithName:(NSString *)attrName {
-    // This method is only used on the iPhone simulator, where normal extended attributes are broken.
-    // See full explanation in the header file.
-
-    if ([attrName length] == 0) {
-        return;
-    }
-
-    // Example:
-    // attrName = "archived"
-    //
-    // "mylog.archived.txt" -> "mylog.txt"
-    // "mylog.archived"     -> "mylog"
-
-    NSArray *components = [[self fileName] componentsSeparatedByString:@"."];
-
-    NSUInteger count = [components count];
-
-    NSUInteger estimatedNewLength = [[self fileName] length];
-    NSMutableString *newFileName = [NSMutableString stringWithCapacity:estimatedNewLength];
-
-    if (count > 0) {
-        [newFileName appendString:components.firstObject];
-    }
-
-    BOOL found = NO;
-
-    NSUInteger i;
-
-    for (i = 1; i < count; i++) {
-        NSString *attr = components[i];
-
-        if ([attrName isEqualToString:attr]) {
-            found = YES;
-        } else {
-            [newFileName appendFormat:@".%@", attr];
-        }
-    }
-
-    if (found) {
-        [self renameFile:newFileName];
-    }
-}
-
-#else /* if TARGET_IPHONE_SIMULATOR */
-
 - (BOOL)hasExtendedAttributeWithName:(NSString *)attrName {
-    const char *path = [filePath UTF8String];
+    const char *path = [filePath fileSystemRepresentation];
     const char *name = [attrName UTF8String];
 
     ssize_t result = getxattr(path, name, NULL, 0, 0, 0);
```

Comment:
With below's comment about "1" or "true", I think we should also check this here. We just need to be careful about backwards compatibility.

Reply:
To maintain backwards compatibility, we could go ahead and have a fast-path, that checks for the value we set (e.g. "1"). If it doesn't match, we can (for now) fall back to the old check, but replace the extended attribute with the new value, so that next time the first check will succeed. With that in place, we can drop legacy code altogether in some future version (whereas now we'd still have to keep it).

Reply:
Fixed in 9e9c7b2